src/modules/settings/CurrencyPage.tsx

```diff
@@ -0,0 +1,41 @@
+import { Box, Container, List, ListItem } from '@mui/material';
+import { HalfTypeNavigation } from '../../shared-components/Navigation/TopNavigation/HalfTypeNavigation';
+import { colors } from 'src/styles/colors';
+import { Currency } from 'src/types/Currency';
+import { ListItemSelect } from '../../shared-components/Card/ListItems/ListItemSelect';
+import { useState } from 'react';
+import { useUserDispatch } from 'src/context/UserContext';
+import { UserActionType } from '../../context/UserContext';
+
+export const CurrencyPage = () => {
+  const [selectedCurrency, setSelectedCurrency] = useState<Currency | null>(
```

Comment:
You don't actually need to use state here. Use data from Context to show selected currency once a user sees component. If user update currency, then update data in Context as you do below.

Make sure that in the context there's a default value set for the currency. Since this is a critical value for the application, this value would be normally set on the backend once user creates and account.

Reply:
fixed